python/foglamp/services/core/scheduler/scheduler.py

```diff
@@ -1123,49 +1137,56 @@ def _schedule_row_to_schedule(cls,
         except KeyError:
             self._logger.info("No Task running for Schedule %s", schedule_id)
 
-        # If a task is running for the schedule, then terminate the process
         if task_id is not None:
             schedule = task_process.schedule
-
-            # TODO: FOGL-356 track the last time TERM was sent to each task
-            task_process.cancel_requested = time.time()
-
-            # Terminate process
-            del self._schedules[schedule.id]
-            try:
-                # KNOWN ISSUE: Does not work well with microservices e.g. COAP etc
-                # TODO: If schedule is a microservice, then deal with shutdown, unregister etc.
-                task_process.process.terminate()
-                task_future = task_process.future
-                if schedule.type != Schedule.Type.STARTUP and task_future.cancel() is True:
-                    await self._wait_for_task_completion(task_process)
+            if schedule.type == Schedule.Type.STARTUP: # If schedule is a service e.g. South services
+                try:
+                    found_services = ServiceRegistry.get(name=schedule.process_name)
+                    service = found_services[0]
+                    if await utils.ping_service(service) is True:
+                        # Shutdown will take care of unregistering the service from core
+                        await utils.shutdown_service(service)
```

Comment:
so now no need to call microservices shutdown in core/server.py? and scheduler stop shall handle it?